src/lex/mod.rs

```diff
@@ -375,281 +220,80 @@ impl Lexer {
         }
     }
     // should only be called at the end of a number. mostly error handling
-    fn parse_exponent(&mut self, hex: bool, mut buf: String) -> Result<f64, LexError> {
+    fn parse_exponent(&mut self, hex: bool) -> Result<(), LexError> {
         let is_digit =
             |c: Option<char>| c.map_or(false, |c| (c as char).is_digit(10) || c == '+' || c == '-');
         if hex {
             if self.match_next('p') || self.match_next('P') {
                 if !is_digit(self.peek()) {
                     return Err(LexError::ExponentMissingDigits);
                 }
-                buf.push('p');
-                buf.push(self.next_char().unwrap() as char);
+                self.next_char();
             }
         } else if self.match_next('e') || self.match_next('E') {
             if !is_digit(self.peek()) {
                 return Err(LexError::ExponentMissingDigits);
             }
-            buf.push('e');
-            buf.push(self.next_char().unwrap() as char);
+            self.next_char();
         }
         while let Some(c) = self.peek() {
             let c = c as char;
             if !(c).is_digit(10) {
                 break;
             }
-            buf.push(c);
             self.next_char();
         }
-        let float: f64 = if hex {
-            let float_literal: hexponent::FloatLiteral = buf.parse()?;
-            float_literal.into()
-        } else {
-            buf.parse()?
-        };
-        let should_be_zero = buf.chars().all(|c| match c {
-            '.' | '+' | '-' | 'e' | 'p' | '0' => true,
-            _ => false,
-        });
-        if float == 0.0 && !should_be_zero {
-            Err(LexError::FloatUnderflow)
-        } else {
-            Ok(float)
-        }
+        Ok(())
     }
     // returns None if there are no digits at the current position
-    fn parse_int(
-        &mut self,
-        mut acc: u64,
-        radix: Radix,
-        buf: &mut String,
-    ) -> Result<Option<u64>, LexError> {
+    fn parse_int(&mut self, radix: Radix) -> Result<Option<()>, LexError> {
         let parse_digit = |c: char| match c.to_digit(16) {
             None => Ok(None),
             Some(digit) if digit < radix.as_u8().into() => Ok(Some(digit)),
             // if we see 'e' or 'E', it's the end of the int, don't treat it as an error
-            // if we see 'b' this could be part of a binary constant (0b1)
-            // if we see 'f' it could be a float suffix
             // we only get this far if it's not a valid digit for the radix, i.e. radix != 16
-            Some(11) | Some(14) | Some(15) => Ok(None),
+            Some(14) => Ok(None),
```

Comment:
Can you explain when the b and f could be there?

Reply:
`0b1110` or `1.431f`. The `e` is for exponents: `1.1e2`

Reply:
The 'b' is consumed at the beginning of a binary constant so that's invalid now, and we don't use `parse_int` for consuming after the decimal place to 'f' would always be invalid here. I think I see what you're doing though - we want to consume all the alpha characters after the digit and report it as an error so there's not a duplicate 'invalid identifier' error.

Reply:
Hmm ... your reasoning makes sense but I remember I needed them for something. 59157678e9a94d1998d2dad4a04f7316d507fa53 and a05e14d6820ee3a5ba22cb5e3338370c0ca38646 are the relevant commits.

Isn't this method also used for the tail parts of floats? I thought it went `fn parse_num { let int = parse_int(); if match_next('.') { let fraction = parse_int(); } }`.

Reply:
For the 'f' suffix, this was added when `parse_int` was used for digits after the decimal point but not anymore - [see here](https://github.com/jyn514/saltwater/commit/524c93d#diff-72dac0bc15578b7a7f1e37d8d65305dbR206).

For the 'b' in binary constants, I'm not sure why it was ever necessary because the 'b' was always removed as far as I can tell.